src/solve.jl

```diff
@@ -110,7 +110,27 @@ function solve(system::InitialValueProblem{<:HybridSystem, <:LazySet{N}},
     # - (discrete) location
     # - (set of) continuous-time reach sets
     # - number of previous jumps
-    waiting_list = [(1, ReachSet{LazySet{N}, N}(X0, zero(N), zero(N)), 0)]
+    inits = length(options[:init]) > 0 ? options[:init] : [(n, X0) for n in states(HS)]
+
+    waiting_list = []
+    for (modeId, x0) in inits
+        mode = HS.modes[modeId]
+        source_invariant = mode.X
+
+        if source_invariant isa HalfSpace
+            # TODO temporary conversion to HPolytope
+            source_invariant = HPolytope([source_invariant])
+        else
+            @assert source_invariant isa HPolytope
```

Comment:
I would just drop this `else` block. If the intersection fails, we will see an error message.